### PR TITLE
Restructure project

### LINKED ISSUE #758: Restructure project
Move everything into dedicated subfolders e.g. `frontend` and `backend`.

Structure the rust part of the project like described here: https://matklad.github.io/2021/08/22/large-rust-workspaces.html

### PR DESCRIPTION
**Pull request by crapStone**
closes #758

- [x] own folders for
  - [x] tauri-app
  - [x] crates that are used by all
  - [x] webserver (src-tauri and src-wasm are also in their own folders)
- [x] README for all subprojects
  - [x] tauri-app
  - [x] wasm
  - [x] webserver
- [x] CI runs again
- [x] everything works like before

### CODE REVIEW

File: .github/workflows/frontend.yml [hunk 1]
```diff
@@ -30,27 +30,20 @@ jobs:
...
 
       - name: Check Formatting
-        run: yarn prettier -c src
+        run: cd frontend && yarn prettier -c src
 
       - name: Eslint
-        run: yarn lint
+        run: cd frontend && yarn lint
 
-      - name: codespell
```
Comment (6543) [inline .github/workflows/frontend.yml:41]: where is it??

File: crates/config/src/handler.rs [hunk 2]
```diff
@@ -265,8 +264,17 @@ fn get_program_data_dir() -> PathBuf {
       .parent()
       .unwrap()
       .to_path_buf();
-    // git repo root
-    program_path.push("../../../");
+    
+      // git repo root
```
Comment (6543) [inline crates/config/src/handler.rs:268]: einrückung ?!?

Comment (6543) [inline crates/config/src/handler.rs:268]:
```suggestion
    // git repo root
```

File: crates/search/src/gestis/mod.rs [hunk 3]
```diff
@@ -73,10 +73,15 @@ impl Gestis {
       }
     }
   }
+
+  #[cfg(feature = "gestis_helper")]
```
Comment (6543) [inline crates/search/src/gestis/mod.rs:77]: add comment to link to contrib script

File: crates/config/src/handler.rs (hunk 2 above)
Comment (crapStone) [reply crates/config/src/handler.rs]: fixed

File: .github/workflows/frontend.yml (hunk 1 above)
Comment (crapStone) [reply .github/workflows/frontend.yml:41]: fixed

File: frontend/.gitignore [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+/coverage
+/libpeerconnection.log
+npm-debug.log
+yarn-error.log
+testem.log
+/typings
+
+# System Files
+.DS_Store
+Thumbs.db
```
Comment (6543) [inline frontend/.gitignore:172]:
```suggestion
```
in root gitignore

File: frontend/.gitignore [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+.settings/
+*.sublime-workspace
+
+# IDE - VSCode
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+.history/*
```
Comment (6543) [inline frontend/.gitignore:158]:
```suggestion
```
in root gitignore

**Comment by 6543** [comment]
`crowdin.example.yml` need fix - new paths!!!

**Review by 6543 — CHANGES_REQUESTED** [review]

File: webserver/rustfmt.toml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+max_width = 120
+hard_tabs = false
+tab_spaces = 2
+newline_style = "Unix"
+use_small_heuristics = "Default"
+reorder_imports = true
+reorder_modules = true
+remove_nested_parens = true
+edition = "2021"
```
Comment (6543) [inline webserver/rustfmt.toml]:
```suggestion
edition = "2018"
```
or was it intentional?

Comment (6543) [inline webserver/rustfmt.toml]:
```suggestion
edition = "2018"
```
or was it intentional?

**Comment by Epsilon02** [comment]
LGTM to the rest.

**Comment by crapStone** [comment]
> `crowdin.example.yml` need fix - new paths!!!

fixed

File: frontend/.gitignore (hunk 4 above)
Comment (crapStone) [reply frontend/.gitignore:172]: done

File: frontend/.gitignore (hunk 5 above)
Comment (crapStone) [reply frontend/.gitignore:158]: done

File: webserver/rustfmt.toml (hunk 6 above)
Comment (crapStone) [reply webserver/rustfmt.toml]: intentional

File: crates/search/src/gestis/mod.rs (hunk 3 above)
Comment (6543) [reply crates/search/src/gestis/mod.rs:77]: @crapStone

Comment (crapStone) [reply crates/search/src/gestis/mod.rs:78]: done

**Review by 6543 — APPROVED** [review]

**Review by Epsilon02 — APPROVED** [review]

**Pull request merged**